### ISSUE TITLE
Low overlap stereo cameras

### ISSUE THREAD

**Issue by mateosss** [darglein/Snake-SLAM#12]
I am trying to run [VR datasets](https://huggingface.co/datasets/mateosss/monado-slam-datasets) with low overlap between the stereo fisheye cameras. And while monocular, monocular-imu seem to track properly, stereo fails likely due to the pre rectification step that occurs when loading datasets. Do you think there can be a reasonable way of making Snake-SLAM run on these datasets without reworking the stereo matcher? SpectacularAI/HybVIO#22 has the same issue

Example frame with overlap in green:

![image](https://github.com/user-attachments/assets/1bf2c784-3661-4ea2-a268-6e59d53a7e3e)

If interested to reproduce the issue you can find the example dataset I am trying to run [here](https://syncandshare.lrz.de/getlink/fi9Hu5M6VreQzHoJWPn2J6/MOO07_for_snakeslam.zip).

Since these datasets use radial-tangential8, a [small modification to saiga](https://github.com/mateosss/saiga/commit/124ddd3cf370bb9bf59ec33a0f978806c2445627) is needed to add the extra k3,k4,k5,k6 parameters.

**Comment by darglein** [comment]
You are right, stereo fisheye is not supported out of the box. It would be definitely more difficult to implement than standard pinhole stereo, because finding the stereo matches is a lot harder. For "soft" fisheye cameras the current system could still work if the rectification gives decent results.

I would suggest you try out the rectification and the stereo matching by its own to see if works well. Unfortunately I don't have much time to try stuff out myself. So I wish you the best of luck with your project :).
